evalml/pipelines/time_series_classification_pipelines.py

```diff
@@ -29,14 +29,15 @@ def __init__(self, parameters, random_seed=0):
         Arguments:
             parameters (dict): Dictionary with component names as keys and dictionary of that component's parameters as values.
                  An empty dictionary {} implies using all default values for component parameters. Pipeline-level
-                 parameters such as gap and max_delay must be specified with the "pipeline" key. For example:
-                 Pipeline(parameters={"pipeline": {"max_delay": 4, "gap": 2}}).
+                 parameters such as date_index, gap, and max_delay must be specified with the "pipeline" key. For example:
+                 Pipeline(parameters={"pipeline": {"date_index": "Date", "max_delay": 4, "gap": 2}}).
             random_seed (int): Seed for the random number generator. Defaults to 0.
         """
         if "pipeline" not in parameters:
-            raise ValueError("gap and max_delay parameters cannot be omitted from the parameters dict. "
+            raise ValueError("date_index, gap, and max_delay parameters cannot be omitted from the parameters dict. "
```

Comment:
So the assumption our code is making after this PR is that whenever a user runs time series automl, they will always have set `date_index`. Correct?

Reply:
Yes that's the assumption we're working under, for all time series problems `date_index` should be specified as a required parameter